scripts/PearlPowderISIS/pearl_calib_factory.py

```diff
@@ -0,0 +1,411 @@
+# pylint: disable=too-many-branches,too-many-statements, superfluous-parens
+
+from mantid.simpleapi import *
+
+
+def get_calibration_dir(cycle, tt_mode, pearl_file_dir):
+    if (cycle == "15_4"):
```

Comment:
Why are you suppressing too-many-branches? the function `get_calibration_dir` has a really high cyclomatic complexity and should be replaced with associative containers.